tests/unit/llm/test_anthropic_llm.py

```diff
@@ -17,14 +17,18 @@
 
 import pytest
 from neo4j_graphrag.llm.anthropic_llm import AnthropicLLM
-
+import sys
 import anthropic
+from typing import Generator
 
 
-def get_mock_anthropic() -> MagicMock:
+@pytest.fixture
+def mock_anthropic() -> Generator[MagicMock, None, None]:
     mock = MagicMock()
     mock.APIError = anthropic.APIError
-    return mock
+
+    with patch.dict(sys.modules, {"anthropic": mock}):
```

Comment:
If I understand correctly, this is overwriting `sys.modules` right? Are we sure we're not deleting anything important like another module?